src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -1389,6 +1389,37 @@ EnsureFunctionOwner(Oid functionId)
 }
 
 
+/*
+ * EnsureDistributedTable error out if the given relation is not a distributed table
+ * with the given message.
+ */
+void
+EnsureDistributedTable(Oid sourceRelationId, const char *errorMessage)
+{
+	if (!IsDistributedTable(sourceRelationId))
+	{
+		ereport(ERROR, (errmsg("%s", errorMessage)));
+	}
+}
+
+
+/*
+ * IsDistributedTable returns true if the given relation is
+ * a distributed table.
+ */
+bool
+IsDistributedTable(Oid relationId)
+{
+	CitusTableCacheEntry *sourceTableEntry = GetCitusTableCacheEntry(relationId);
+	if (sourceTableEntry == NULL)
+	{
+		return false;
+	}
+	char sourceDistributionMethod = sourceTableEntry->partitionMethod;
+	return sourceDistributionMethod == DISTRIBUTE_BY_HASH;
```

Comment:
this should consider other distributed table types as well.

Reply:
I have changed this to `IsHashDistributedTable`